File: Cpp/856_Score_of_Parentheses.cpp
class Solution {
public:
    int scoreOfParentheses(string s) {
        int result = 0;
        int open = 0;
        
        for(int i = 0; i < s.size(); ++i) {
            if(s[i] == '(') {
                ++open;
            } else {
                --open;
                if(s[i - 1] == '(') {
                    result += pow(2, open);
                }
            }
        }
        
        return result;
    }
};

File: Kotlin/856_Score_of_Parentheses.kt
import kotlin.math.*

class Solution {
    fun scoreOfParentheses(s: String): Int {
        var result = 0.0
        var open = 0
        
        for(i in s.indices) {
            if(s[i] == '(') {
                ++open
            } else {
                --open
                if(s[i - 1] == '(') {
                    result += 2.0.pow(open)
                }
            }
        }
        
        return result.toInt()
    }
}